src/sync/sync_session.hpp

```diff
@@ -275,6 +275,7 @@ class SyncSession : public std::enable_shared_from_this<SyncSession> {
     size_t m_death_count = 0;
 
     SyncConfig m_config;
+    util::Optional<std::array<char, 64>> m_realm_encryption_key;
```

Comment:
This sorta feels like it should be a member of `SyncConfig`, since that holds all of the information `SyncSession` needs. The downside there is that the binding would need to configure the encryption key in both `Realm::Config` and `SyncConfig` when opening a synced Realm.